### PR TITLE
[WIP] Implement Proposals.RSlice

### PR DESCRIPTION
**Pull request by SaranjeetKaur**

### CODE REVIEW

File: src/proposals/Proposals.jl [hunk 1]
```diff
@@ -248,5 +249,140 @@ function (prop::RStagger)(rng::AbstractRNG,
...
+                if s < 0    # left
+                    u_l = u_prop
+                elseif s > 0    # right
+                    u_r = u_prop
+                else     # if `s = 0` something has gone wrong
+                    error("Random slice sampler has failed to find a valid point.")
+                end
+            end
+        end # end of sample within limits while       
+        end # end of random slice sampling loop
```
Comment (mileslucas) [inline src/proposals/Proposals.jl:379]: all of this code is exactly the same as the other slice sampling method. I would strongly suggest factoring this common functionality out into a separate method that just takes a vector "axis" and axlength and whatever other things and ran this code! saves a lot of space and keeps testing easier.

File: src/proposals/Proposals.jl [hunk 2]
```diff
@@ -248,5 +249,140 @@ function (prop::RStagger)(rng::AbstractRNG,
...
+                       point::AbstractVector,
+                       logl_star,
+                       bounds::AbstractBoundingSpace,
+                       loglike,
+                       prior_transform;
+                       kwargs...)
+    # setup
+    n = length(point)
+    nc = nexpand = ncontract = 0
+    fscale = []
```
Comment (mileslucas) [inline src/proposals/Proposals.jl:278]:
```suggestion
```

File: src/proposals/Proposals.jl (hunk 1 above)
Comment (SaranjeetKaur) [reply src/proposals/Proposals.jl]: done

**Comment by mileslucas** [comment]
The way you've decoupled the slice sampling code has introduced quite a few new bugs, most of which are related to not properly recognizing which variables are used and where they are used. I've gone ahead and fixed them and pushed them since it would be easier than trying to iterate through PR comments.

**Pull request merged**